File: app/src/main/java/net/emojiparty/android/jishotomo/ui/dialogs/CallbackDialog.java
package net.emojiparty.android.jishotomo.ui.dialogs;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.appcompat.app.AlertDialog;
import net.emojiparty.android.jishotomo.R;

public class CallbackDialog extends DialogFragment {
  private OnConfirm callback;
  private int confirmText;
  private int dialogText;

  public CallbackDialog(OnConfirm callback, int dialogText, int confirmText) {
    this.callback = callback;
    this.dialogText = dialogText;
    this.confirmText = confirmText;
  }

  public interface OnConfirm {
    void proceed();
  }


  // https://developer.android.com/guide/topics/ui/dialogs
  @NonNull @Override public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
    AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
    builder.setMessage(dialogText)
        .setPositiveButton(confirmText, (DialogInterface dialog, int id) -> {
          callback.proceed();
        })
        .setNegativeButton(R.string.cancel, null);
    return builder.create();
  }
}

File: app/src/main/java/net/emojiparty/android/jishotomo/ui/dialogs/CallbackDialog.kt
package net.emojiparty.android.jishotomo.ui.dialogs

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog.Builder
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentManager
import net.emojiparty.android.jishotomo.R.string

open class CallbackDialog(
  private val dialogText: Int,
  private val confirmText: Int
) : DialogFragment() {

  // https://developer.android.com/guide/topics/ui/dialogs
  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
    return Builder(requireActivity())
      .setMessage(dialogText)
      .setPositiveButton(confirmText) { _: DialogInterface?, _: Int -> onConfirm() }
      .setNegativeButton(string.cancel, null)
      .create()
  }

  open fun onConfirm() { }

  protected open val dialogTag: String = ""

  fun show(fragmentManager: FragmentManager) {
    show(fragmentManager, dialogTag)
  }
}
